platform/kube/controller.go

```diff
@@ -131,6 +134,25 @@ func NewController(client *Client, options ControllerOptions) *Controller {
 			func(opts meta_v1.ListOptions) (watch.Interface, error) {
 				return client.client.ExtensionsV1beta1().Ingresses(options.Namespace).Watch(opts)
 			})
+
+		// Ingress sync mode -> ingress status sync mode. The ingress status is synced in two cases:
+		// 1. default ingress class provided and no annotation on ingress resource
+		// 2. specific ingress class provided and same annotation on ingress resource
+		var ingressClass, defaultIngressClass string
+		switch options.IngressSyncMode {
+		case IngressStrict:
+			ingressClass = options.IngressClass
+		case IngressDefault:
+			defaultIngressClass = options.IngressClass
```

Comment:
Good catch. The meaning in ingress syncer doesn't match up with that:
```go
// IsValid returns true if the given Ingress either doesn't specify
// the ingress.class annotation, or it's set to the configured in the
// ingress controller.
func IsValid(ing *extensions.Ingress, controller, defClass string) bool {
	ingress, err := parser.GetStringAnnotation(IngressKey, ing)
	if err != nil && !errors.IsMissingAnnotations(err) {
		glog.Warningf("unexpected error reading ingress annotation: %v", err)
	}

	// we have 2 valid combinations
	// 1 - ingress with default class | blank annotation on ingress
	// 2 - ingress with specific class | same annotation on ingress
	//
	// and 2 invalid combinations
	// 3 - ingress with default class | fixed annotation on ingress
	// 4 - ingress with specific class | different annotation on ingress
	if ingress == "" && controller == defClass {
		return true
	}

	return ingress == controller
}
```
That would be 3, which is considers invalid.